ui/components/multichain-accounts/add-wallet-modal/add-wallet-modal.tsx

```diff
@@ -46,28 +55,42 @@ export const AddWalletModal: React.FC<AddWalletModalProps> = ({
   ...props
 }) => {
   const t = useI18nContext();
+  const history = useHistory();
 
   const walletOptions: WalletOption[] = [
     {
       id: 'import-wallet',
       titleKey: 'importAWallet',
       iconName: IconName.Wallet,
+      route: IMPORT_SRP_ROUTE,
     },
     {
       id: 'import-account',
       titleKey: 'importAnAccount',
       iconName: IconName.Download,
+      route: ADD_WALLET_PAGE_ROUTE,
     },
     {
       id: 'hardware-wallet',
       titleKey: 'addAHardwareWallet',
       iconName: IconName.Hardware,
+      route: CONNECT_HARDWARE_ROUTE,
     },
   ];
 
-  const handleOptionClick = () => {
-    // TODO Handle optionId selection logic here in subsequent PR
+  const handleOptionClick = (option: WalletOption) => {
     onClose?.();
+
+    // Hardware wallet connections require expanded view
+    if (option.id === 'hardware-wallet') {
+      if (getEnvironmentType() === ENVIRONMENT_TYPE_POPUP) {
+        global.platform.openExtensionInBrowser?.(option.route);
+      } else {
+        history.push(option.route);
+      }
```

Comment:
This uses the same logic as the account menu https://github.com/MetaMask/metamask-extension/blob/c1e0e51c4850d298e2960d67ca21bdbebf04a0df/ui/components/multichain/account-menu/account-menu.tsx#L581-L587